packages/smooth_app/lib/cards/product_cards/question_card.dart

```diff
@@ -377,14 +381,16 @@ Future<void> saveAnswer({
   required String? insightId,
   required InsightAnnotation insightAnnotation,
 }) async {
-  // TODO(jasmeet): Send answer to the Backend.
-  // TODO(jasmeet): Fix for iOS. https://pub.dev/packages/device_info
-  // DeviceInfoPlugin deviceInfo = DeviceInfoPlugin();
-  // AndroidDeviceInfo androidInfo = await deviceInfo.androidInfo;
-  //
-  // final Status status = await OpenFoodAPIClient.postInsightAnnotation(
-  //   insightId,
-  //   insightAnnotation,
-  //   deviceId: androidInfo.id,
-  // );
+  final DeviceInfoPlugin deviceInfoPlugin = DeviceInfoPlugin();
+  String? deviceId;
+  if (Platform.isAndroid) {
+    deviceId = (await deviceInfoPlugin.androidInfo).androidId;
+  } else if (Platform.isIOS) {
+    deviceId = (await deviceInfoPlugin.iosInfo).identifierForVendor;
+  }
+  await OpenFoodAPIClient.postInsightAnnotation(
+    insightId,
+    insightAnnotation,
+    deviceId: deviceId,
```

Comment:
Not sure of the added value of `deviceId`.

Reply:
Discussion in the other comment